lib/prelude/preloaded.rb

```diff
@@ -0,0 +1,32 @@
+module Prelude
+  class ValueNotPreloaded < StandardError; end
+
+  class Preloaded
```

Comment:
Good call, done.

Thinking along these lines I also renamed the `Preloaded#fetch` method to `#fetch_preloaded_value` so that it's less likely to collide with a prelude method.